crates/bevy_ecs/src/schedule/system_graph.rs

```diff
@@ -0,0 +1,264 @@
+use crate::schedule::{SystemDescriptor, SystemLabel, SystemSet};
+use std::{
+    collections::HashMap,
+    fmt::Debug,
+    sync::{
+        atomic::{AtomicU32, Ordering},
+        Arc, Mutex,
```

Comment:
While I prefer the `Rc`/`RefCell` approach, if we do use a `Mutex` we should use `parking_lot::Mutex`

Reply:
Should have checked the dependencies, didn't want to assume it was available.